### PR TITLE
Respect CFLAGS and CPPFLAGS from environment

### PR DESCRIPTION
**Pull request by okias**
Debian and Mobian would be very grateful, if build-system would be able to accept distribution options (in case of Debian mostly focus on hardening) by default without need of patching.

Follow up of:
 - https://github.com/minetest/minetest/pull/12799
 - https://github.com/minetest/irrlicht/pull/134

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -37,6 +37,7 @@ if(NOT (CMAKE_BUILD_TYPE IN_LIST SUPPORTED_BUILD_TYPES))
 			"Unknown CMAKE_BUILD_TYPE: ${CMAKE_BUILD_TYPE}. Options are: ${SUPPORTED_BUILD_TYPES}.")
 endif()
 
+set(CMAKE_C_FLAGS "${CMAKE_CFLAGS} $ENV{CFLAGS} $ENV{CPPFLAGS}")
```
Comment (sfan5) [inline src/CMakeLists.txt:40]: is `CMAKE_CFLAGS` a typo?

Comment (okias) [reply src/CMakeLists.txt]: definitely, fixed now

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -844,7 +845,7 @@ else()
 		endif()
 	endif()
 
-	set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG ${RELEASE_WARNING_FLAGS} ${OTHER_FLAGS} -pipe -funroll-loops -O3 -fomit-frame-pointer")
+set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG ${RELEASE_WARNING_FLAGS} ${OTHER_FLAGS} -pipe -funroll-loops -O3 -fomit-frame-pointer $ENV{CPPFLAGS}")
```
Comment (sfan5) [inline src/CMakeLists.txt:848]:
```suggestion
	set(CMAKE_CXX_FLAGS_RELEASE "-DNDEBUG ${RELEASE_WARNING_FLAGS} ${OTHER_FLAGS} -pipe -funroll-loops -O3 -fomit-frame-pointer $ENV{CPPFLAGS}")
```

**Comment by okias** [comment]
Btw. while it has characteristic of feature from game perspective, I would say it's more like bug-fix for distro packaging :)

**Review by sfence — CHANGES_REQUESTED** [review]
PR works as expected.
It should be documented somewhere somehow.
Probably `doc/compiling/README.md `?

**Comment by okias** [comment]
Thank you for ping, feel free to reopen when suitable.

**Comment by sfence** [comment]
@okias Feel free to reopen PR when you've updated the documentation.

**Comment by okias** [comment]
I updated doc before the comment, I CANNOT reopen.

**Comment by grorp** [comment]
Since you force-pushed, nobody can reopen this PR AFAICT 🙃

**Comment by okias** [comment]
you could just ping me instead closing. Anyway follow-up here: https://github.com/minetest/minetest/pull/15230

**Pull request closed without merging**